### PR TITLE
VFX2023 merge to main

### PR DESCRIPTION
**Pull request by tbabiin**
### Linked issues
N/A

### Summarize your change.
Merging VFX2023 to main and adding a commit for fixing py-interp

### Describe the reason for the change.

RV is updating their third-party toolchains

### Describe what you have tested and on which operating system.

Mac, Windows and Linux Rocky 8 and Rocky 9

### Add a list of changes, and note any that might need special attention during the review.
- Added a commit to fix py-interp: removing Py_Initialize as it is called by Py_Main. Removing PySys_SetArgvEx as it doesn't work when removing Py_Initialize but the new PyConfig works to set Py[...]SetArgv.

### If possible, provide screenshots.

### CODE REVIEW

**Comment by tbabiin** [comment]
Only [46b8289](https://github.com/AcademySoftwareFoundation/OpenRV/pull/294/commits/46b82895bda4ba5338f8f6b1790944508ab80277) needs to be reviewed, the rest is a branch merge.

**Review by geffrak — APPROVED** [review]

File: src/build/make_pyside.py [hunk 1]
```diff
@@ -222,7 +222,7 @@ def build() -> None:
     # PySide2 v5.15.2.1 builds with errors on Windows using Visual Studio 2019.
     # We force Visual Studio 2017 here to make it build without errors.
     if platform.system() == "Windows":
-        source_widows_msvc_env("2017")
+        source_widows_msvc_env("2022")
```
Comment (geffrak) [inline src/build/make_pyside.py:225]: Do we still need that now that it's the same version as the rest of the build? Not asking for a change here.

Comment (tbabiin) [reply src/build/make_pyside.py:225]: Maybe we could test after merging. The scenario I'm not sure about is what happens with a computer that has: (VS2017) + VS2019 + VS2022; if we don't source, which one does PySide decides to pick. 🤔

**Review by bernie-laberge — APPROVED** [review]

**Review by rogernelson — APPROVED** [review]

**Comment by tbabiin** [comment]
Forced-pushed again for another rebase to fix a conflict in make_pyside: it wasn't a real conflict, only that one of the line that I changed had a remote change for the line just before thus, a conflict for a remote change on the previous line of my change. 🤷‍♂️

**Pull request merged**